### PR TITLE
Fixed multi fit bug in muon analysis when changing fit function

### LINKED ISSUE #22495: Muon multiple fitting bug
Open muon analysis load some HIFI data. Go to data analysis (multiple fitting) and do a fit. Remove the fitting function and then add a different one. When you try to fit it fails. 

Reported by @keeeto

### PR DESCRIPTION
**Pull request by AnthonyLim23**
In muon analysis the multiple fitting would crash if the user changed the fitting function or used undo fit

**To test:**
Open muon analysis
Load some data
Go to settings and make sure multiple fitting is ticked
Go to data analysis
Add a fitting function and do a fit
Remove the fitting function
Add a different fitting function
Do another fit (this is where it would crash)

Do undo fit (this would also crash)

Go to settings tab and untick multiple fit
Go to data analysis
Try a few different fits.

Fixes #22495. 

**Release Notes** 

In release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by keeeto — APPROVED** [review]
This works as described and removes the crashing behaviour that I previously experienced.

**Review by mantid-roman — CHANGES_REQUESTED** [review]
Sorry, but this is the wrong place to fix the bug.

**Comment by mantid-roman** [comment]
I am looking for a better way of fixing it.

**Comment by mantid-roman** [comment]
I think a slightly better way of fixing the bug is:

1. Add a public method to `QtAbstractPropertyManager` in `qtpropertybrowser.cpp`:
  ```
 bool QtAbstractPropertyManager::hasProperty(const QtProperty* prop) const {
    return d_ptr->m_properties.contains(prop);
  }
  ```
2. Modify the `clearError` method:
```
void ParameterPropertyManager::clearError(QtProperty *property) {
  m_errors.remove(property);
  if (hasProperty(property)) {
    emit propertyChanged(property);
    updateTooltip(property);
  }
}
```

**Comment by AnthonyLim23** [comment]
@mantid-roman that code does not work as the `m_properties.contains` expects a set of` QtProperty` (not the pointer). I have had a quick look but have not had much look in getting it to work.

**Comment by mantid-roman** [comment]
This signature should work: 
```
bool hasProperty(QtProperty*const prop) const
```

**Comment by keeeto** [comment]
I'm afraid it seems like the bug is still present in this update.

File: qt/widgets/common/src/QtPropertyBrowser/ParameterPropertyManager.cpp [hunk 1]
```diff
@@ -77,9 +77,9 @@ void ParameterPropertyManager::setDescription(QtProperty *property,
  * @param property :: Property to clear error for
  */
 void ParameterPropertyManager::clearError(QtProperty *property) {
-  m_errors.remove(property);
-  emit propertyChanged(property);
-  updateTooltip(property);
+		m_errors.remove(property);
```
Comment (mantid-roman) [inline qt/widgets/common/src/QtPropertyBrowser/ParameterPropertyManager.cpp]: A test for `hasProperty(property)` is missing.

Comment (AnthonyLim23) [reply qt/widgets/common/src/QtPropertyBrowser/ParameterPropertyManager.cpp]: ah yes. Thank you

**Comment by keeeto** [comment]
This now works again 👍

**Review by mantid-roman — APPROVED** [review]

**Pull request merged**